In [2]:
import geopandas as gpd
import pandas as pd
import numpy as np

In [3]:
# read in KNN-Access data analysis results 
df = gpd.read_file('../data/aziz-final/Seattle Output Files/origin_record.geojson')

In [4]:
df.head()

Baseline  UG1 Network  UG2 Network  UG3 Network  UG1 Node  UG2 Node  \
0  5.779995     2.214172     0.728205     0.000000  5.474603  0.455625   
1  5.779995     2.214172     0.728205     0.000000  5.474603  0.455625   
2  9.914821     9.753530     7.781496     5.035237  9.895874  9.858677   
3  9.914821     9.753530     7.781496     5.035237  9.895874  9.858677   
4  9.914821     9.753530     7.781496     5.035237  9.895874  9.858677   

   UG3 Node  UG1 Elevation  UG2 Elevation  UG3 Elevation  UG1 Network+Node  \
0  0.000000       5.761933       5.727985       5.608296          1.740703   
1  0.000000       5.761933       5.727985       5.608296          1.740703   
2  9.825177       9.851188       9.740542       9.728332          9.741421   
3  9.825177       9.851188       9.740542       9.728332          9.741421   
4  9.825177       9.851188       9.740542       9.728332          9.741421   

   UG2 Network+Node  UG3 Network+Node  UG1 Network+Node+Elevation  \
0          0.000000          0.000000                    1.387189   
1          0.000000          0.000000                    1.387189   
2          7.866494          5.037969                    9.686386   
3          7.866494          5.037969                    9.686386   
4          7.866494          5.037969                    9.686386   

   UG2 Network+Node+Elevation  UG3 Network+Node+Elevation  \
0                    0.000000                    0.000000   
1                    0.000000                    0.000000   
2                    7.400217                    4.966901   
3                    7.400217                    4.966901   
4                    7.400217                    4.966901   

                         geometry  
0  POINT (1260267.129 243780.167)  
1  POINT (1260267.129 243780.167)  
2  POINT (1260483.365 243274.057)  
3  POINT (1260483.365 243274.057)  
4  POINT (1260483.365 243274.057)

# Differences between Baseline and UG1, UG2, UG3

In [48]:
df_difference= df.copy()

In [49]:
df_difference.head()

Baseline  UG1 Network  UG2 Network  UG3 Network  UG1 Node  UG2 Node  \
0  5.779995     2.214172     0.728205     0.000000  5.474603  0.455625   
1  5.779995     2.214172     0.728205     0.000000  5.474603  0.455625   
2  9.914821     9.753530     7.781496     5.035237  9.895874  9.858677   
3  9.914821     9.753530     7.781496     5.035237  9.895874  9.858677   
4  9.914821     9.753530     7.781496     5.035237  9.895874  9.858677   

   UG3 Node  UG1 Elevation  UG2 Elevation  UG3 Elevation  UG1 Network+Node  \
0  0.000000       5.761933       5.727985       5.608296          1.740703   
1  0.000000       5.761933       5.727985       5.608296          1.740703   
2  9.825177       9.851188       9.740542       9.728332          9.741421   
3  9.825177       9.851188       9.740542       9.728332          9.741421   
4  9.825177       9.851188       9.740542       9.728332          9.741421   

   UG2 Network+Node  UG3 Network+Node  UG1 Network+Node+Elevation  \
0          0.000000          0.000000                    1.387189   
1          0.000000          0.000000                    1.387189   
2          7.866494          5.037969                    9.686386   
3          7.866494          5.037969                    9.686386   
4          7.866494          5.037969                    9.686386   

   UG2 Network+Node+Elevation  UG3 Network+Node+Elevation  \
0                    0.000000                    0.000000   
1                    0.000000                    0.000000   
2                    7.400217                    4.966901   
3                    7.400217                    4.966901   
4                    7.400217                    4.966901   

                         geometry  
0  POINT (1260267.129 243780.167)  
1  POINT (1260267.129 243780.167)  
2  POINT (1260483.365 243274.057)  
3  POINT (1260483.365 243274.057)  
4  POINT (1260483.365 243274.057)

In [50]:
df_difference.loc[:,'UG1_diff'] =  df_difference['Baseline'] - df_difference['UG1 Network+Node+Elevation']
df_difference.loc[:,'UG2_diff'] =  df_difference['Baseline'] - df_difference['UG2 Network+Node+Elevation']
df_difference.loc[:,'UG3_diff'] =  df_difference['Baseline'] - df_difference['UG3 Network+Node+Elevation']

In [53]:
df_difference.sort_values(by='UG1_diff').head()

Baseline  UG1 Network  UG2 Network  UG3 Network   UG1 Node   UG2 Node  \
8072   10.308456    10.743324    12.098166    12.019551  10.502046  10.703707   
8073   10.308456    10.743324    12.098166    12.019551  10.502046  10.703707   
8133    9.782319    10.221341    11.538229    10.196442   9.727513  10.070710   
7691   10.529486    12.051083    12.062269    12.143656  10.557400  10.686363   
40263  11.055004    11.531872    11.746219     7.485552  11.063871  11.091083   

        UG3 Node  UG1 Elevation  UG2 Elevation  UG3 Elevation  \
8072   10.774897      10.300858      10.246919      10.146443   
8073   10.774897      10.300858      10.246919      10.146443   
8133   10.070710       9.765491       9.745816       9.698414   
7691   10.778946      10.477464      10.420799      10.317854   
40263  11.091083      11.079159      11.002118      11.025171   

       UG1 Network+Node  UG2 Network+Node  UG3 Network+Node  \
8072          10.726006         12.280040         11.724458   
8073          10.726006         12.280040         11.724458   
8133          11.276601         11.501912         10.196442   
7691          12.032906         12.245251         11.680980   
40263         11.564818         11.190323          7.485552   

       UG1 Network+Node+Elevation  UG2 Network+Node+Elevation  \
8072                    12.035329                   12.183715   
8073                    12.035329                   12.183715   
8133                    11.246284                   11.473403   
7691                    11.992378                   12.150416   
40263                   12.516534                   10.761929   

       UG3 Network+Node+Elevation                        geometry  UG1_diff  \
8072                    11.729568  POINT (1259978.431 235906.576) -1.726873   
8073                    11.729568  POINT (1259978.431 235906.576) -1.726873   
8133                    10.069378  POINT (1260287.376 235697.925) -1.463965   
7691                    11.687654  POINT (1259856.944 235927.639) -1.462892   
40263                    6.494224  POINT (1262784.703 231433.477) -1.461530   

       UG2_diff  UG3_diff  
8072  -1.875259 -1.421112  
8073  -1.875259 -1.421112  
8133  -1.691084 -0.287059  
7691  -1.620930 -1.158168  
40263  0.293075  4.560780

In [54]:
df_difference[df_difference['UG1_diff'] < 0].count().iloc[0]

5305

In [55]:
df_difference[df_difference['UG2_diff'] < 0].count().iloc[0]

1430

In [56]:
df_difference[df_difference['UG3_diff'] < 0].count().iloc[0]

297

In [58]:
print (5305 + 1430 + 297, round((5305 + 1430 + 297)/df_difference.shape[0], 3))

7032 0.143


Conclusion: There are 7,032 (14.3%) points where the KNN-Access Score for Network+Node+Elevation is lower than the Baseline, which are incorrect.

# Accessibility Gap V1

Accessibility Gap = UG_X Elevation - UG_X Network+Node+Elevation 

In [18]:
df_gap_1= df.copy()

In [19]:
df_gap_1.loc[:, 'UG1_gap'] = df_gap_1['UG1 Elevation'] - df_gap_1['UG1 Network+Node+Elevation']
df_gap_1.loc[:, 'UG2_gap'] = df_gap_1['UG2 Elevation'] - df_gap_1['UG2 Network+Node+Elevation']
df_gap_1.loc[:, 'UG3_gap'] = df_gap_1['UG3 Elevation'] - df_gap_1['UG3 Network+Node+Elevation']

In [24]:
df_gap_1[df_gap_1['UG1_gap'] < 0].count().iloc[0]

7977

In [25]:
df_gap_1[df_gap_1['UG2_gap'] < 0].count().iloc[0]

1927

In [28]:
df_gap_1[df_gap_1['UG3_gap'] < 0].count().iloc[0]

570

In [36]:
print(7977 + 1927 + 570, round((7977 + 1927 + 570) / len(df_gap_1), 3))

10474 0.214


Conclusion: There are 10,474 (21.4%) points where the accessible gap is negative, which are incorrect.


# Accessibility Gap V2

Accessibility Gap = Baseline - UG_X Network+Node

In [40]:
df_gap_2= df.copy()

In [41]:
df_gap_2.loc[:, 'UG1_gap'] = df_gap_2['Baseline'] - df_gap_2['UG1 Network+Node']
df_gap_2.loc[:, 'UG2_gap'] = df_gap_2['Baseline'] - df_gap_2['UG2 Network+Node']
df_gap_2.loc[:, 'UG3_gap'] = df_gap_2['Baseline'] - df_gap_2['UG3 Network+Node']

In [43]:
# accessibility gap < 0 for UG1
df_gap_2[df_gap_2['UG1_gap'] < 0].count().iloc[0]

7762

In [44]:
# accessibility gap < 0 for UG2
df_gap_2[df_gap_2['UG2_gap'] < 0].count().iloc[0]

2699

In [46]:
# accessibility gap < 0 for UG3
df_gap_2[df_gap_2['UG3_gap'] < 0].count().iloc[0]

583

In [47]:
print(7762 + 2699 + 583, round((7762 + 2699 + 583) / len(df_gap_2), 3))

11044 0.225


Conclusion: There are 11,044 (22.5%) points where the accessible gap is negative, which are incorrect.